sklearn/cluster/dbscan_.py

```diff
@@ -125,6 +125,10 @@ def dbscan(X, eps=0.5, min_samples=5, metric='minkowski', metric_params=None,
         X_mask = X.data <= eps
         masked_indices = astype(X.indices, np.intp, copy=False)[X_mask]
         masked_indptr = np.cumsum(X_mask)[X.indptr[1:] - 1]
+
+        if X.indptr[0] == X.indptr[1] == 0:  # check if first row is all zero
+            masked_indptr[0] = 0
```

Comment:
For the case where multiple initial rows are all zero, this index needs to be `[:np.argmin(masked_indptr)]`.

The other way of doing it is replacing `np.cumsum(X_mask)[X.indptr[1:] - 1]` with `np.concatenate([0, np.cumsum(X_mask)][X.indptr[1:]]`. Choose whichever you feel is more intuitive (the latter is a bit less efficient, but it pales in comparison to DBSCAN complexity overall).

Please add a test and a bug fix entry in `whats_new.rst`. Thanks.